java/ql/src/experimental/Security/CWE/CWE-470/UnsafeReflection.ql

```diff
@@ -0,0 +1,49 @@
+/**
+ * @name Use of Externally-Controlled Input to Select Classes or Code ('Unsafe Reflection')
+ * @description Use external input with reflection function to select the class or code to
+ *              be used, which brings serious security risks.
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id java/unsafe-reflection
+ * @tags security
+ *       external/cwe/cwe-470
+ */
+
+import java
+import UnsafeReflectionLib
+import semmle.code.java.dataflow.FlowSources
+import DataFlow::PathGraph
+
+class ContainsSanitizer extends DataFlow::BarrierGuard {
```

Comment:
All public elements: make private or add docstring

Reply:
ok

Reply:
I have changed to private